flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -114,6 +115,10 @@ public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
                 ReconciliationUtils.updateForReconciliationError(flinkApp, validationError.get());
                 return ReconciliationUtils.toUpdateControl(
                         operatorConfiguration, originalCopy, flinkApp, false);
+            } else if (flinkApp.getSpec()
+                    .getFlinkConfiguration()
+                    .containsKey(ExecutionCheckpointingOptions.CHECKPOINTING_INTERVAL.key())) {
+                FlinkUtils.setCheckpointingInterval(originalCopy.getSpec().getFlinkConfiguration());
```

Comment:
We should simply add this logic to the FlinkConfigBuilder instead of modifying the spec